transcendence-app/src/game/stats/infrastructure/db/postgres/user-level.postgres.repository.ts

```diff
@@ -94,20 +93,20 @@ export class UserLevelPostgresRepository
         text: `SELECT *
                FROM ${table.GAME}
                WHERE ${gameKeys.PLAYERONEUSERNAME} = $1
-                 OR ${gameKeys.PLAYERTWOUSERNAME} = $1;`,
+                  OR ${gameKeys.PLAYERTWOUSERNAME} = $1;`,
         values: [username],
       });
     }
     return gamesData ? gamesData.map((game) => new Game(game)) : null;
   }
 
-  async getWinGameRatio(
+  async getGameResults(
```

Comment:
suggestion: I think you already wrote this in another PR. We could do this with SQL with a query like:

```ts
  async getGameResults(
    username: string,
    gameMode?: GameMode,
  ): Promise<GameStats | null> {
    let gamesStatsData: GameStatsData[] | null = null;
    if (gameMode) {
      gamesStatsData = await makeQuery<GameStatsData>(this.pool, {
        text: `
SELECT
    COUNT(CASE WHEN (${gameKeys.PLAYERONEUSERNAME} = $1 AND ${gameKeys.PLAYERONESCORE} > ${gameKeys.PLAYERTWOSCORE}) OR (${gameKeys.PLAYERTWOUSERNAME} = $1 AND ${gameKeys.PLAYERTWOSCORE} > ${gameKeys.PLAYERONESCORE}) THEN 1 END) AS wins,
    COUNT(CASE WHEN (${gameKeys.PLAYERONEUSERNAME} = $1 AND ${gameKeys.PLAYERONESCORE} < ${gameKeys.PLAYERTWOSCORE}) OR (${gameKeys.PLAYERTWOUSERNAME} = $1 AND ${gameKeys.PLAYERTWOSCORE} < ${gameKeys.PLAYERONESCORE}) THEN 1 END) AS losses,
    COUNT(CASE WHEN (${gameKeys.PLAYERONEUSERNAME} = $1 OR ${gameKeys.PLAYERTWOUSERNAME} = $1) AND ${gameKeys.PLAYERONESCORE} = ${gameKeys.PLAYERTWOSCORE} THEN 1 END) AS draws
FROM ${table.GAME}
WHERE (${gameKeys.PLAYERONEUSERNAME} = $1 OR ${gameKeys.PLAYERTWOUSERNAME} = $1)
AND ${gameKeys.GAMEMODE} = $2;
        `,
        values: [username, gameMode],
      });
    } else {
      gamesStatsData = await makeQuery<GameStatsData>(this.pool, {
        text: `
SELECT
    COUNT(CASE WHEN (${gameKeys.PLAYERONEUSERNAME} = $1 AND ${gameKeys.PLAYERONESCORE} > ${gameKeys.PLAYERTWOSCORE}) OR (${gameKeys.PLAYERTWOUSERNAME} = $1 AND ${gameKeys.PLAYERTWOSCORE} > ${gameKeys.PLAYERONESCORE}) THEN 1 END) AS wins,
    COUNT(CASE WHEN (${gameKeys.PLAYERONEUSERNAME} = $1 AND ${gameKeys.PLAYERONESCORE} < ${gameKeys.PLAYERTWOSCORE}) OR (${gameKeys.PLAYERTWOUSERNAME} = $1 AND ${gameKeys.PLAYERTWOSCORE} < ${gameKeys.PLAYERONESCORE}) THEN 1 END) AS losses,
    COUNT(CASE WHEN (${gameKeys.PLAYERONEUSERNAME} = $1 OR ${gameKeys.PLAYERTWOUSERNAME} = $1) AND ${gameKeys.PLAYERONESCORE} = ${gameKeys.PLAYERTWOSCORE} THEN 1 END) AS draws
FROM ${table.GAME}
WHERE (${gameKeys.PLAYERONEUSERNAME} = $1 OR ${gameKeys.PLAYERTWOUSERNAME} = $1);
        `,
        values: [username],
      });
    }
    return gamesStatsData && gamesStatsData.length
      ? new GameStats(gamesStatsData[0])
      : null;
  }
```

Reply:
This looks very nice. Will do.

Reply:
done. Thank you!